src/ApiWrapper.php

```diff
@@ -264,6 +278,12 @@ public function fetchDocument($filename = false)
             $fields['doc[prince_options][baseurl]'] = $this->baseurl;
         }
 
+        if (!empty($this->document_prince_options) && is_array($this->document_prince_options)) {
```

Comment:
I don't think you need the `!empty` check here. Also consider moving the array validation to `ApiWrapper.php`. `ApiWrapper` is responsible for making sure the data is valid. This class is for making sure the request is properly formatted.